pkg/apis/machineconfiguration.openshift.io/v1/types.go

```diff
@@ -491,3 +491,109 @@ type KubeletConfigList struct {
 
 	Items []KubeletConfig `json:"items"`
 }
+
+// CustomResourceDefinition for ContainerRuntimeConfig
+// apiVersion: apiextensions.k8s.io/v1beta1
+// kind: CustomResourceDefinition
+// metadata:
+//   # name must match the spec fields below, and be in the form: <plural>.<group>
+//   name: containerruntimeconfigs.machineconfiguration.openshift.io
+// spec:
+//   # group name to use for REST API: /apis/<group>/<version>
+//   group: machineconfiguration.openshift.io
+//   # list of versions supported by this CustomResourceDefinition
+//   versions:
+//     - name: v1
+//       # Each version can be enabled/disabled by Served flag.
+//       served: true
+//       # One and only one version must be marked as the storage version.
+//       storage: true
+//   # either Namespaced or Cluster
+//   scope: Cluster
+//   names:
+//     # plural name to be used in the URL: /apis/<group>/<version>/<plural>
+//     plural: containerruntimeconfigs
+//     # singular name to be used as an alias on the CLI and for display
+//     singular: containerruntimeconfig
+//     # kind is normally the CamelCased singular type. Your resource manifests use this.
+//     kind: ContainerRuntimeConfig
+//     # shortNames allow shorter string to match your resource on the CLI
+//     shortNames: ['ctrcfg']
+//
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ContainerRuntimeConfig describes a customized Container Runtime configuration.
+type ContainerRuntimeConfig struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   ContainerRuntimeConfigSpec   `json:"spec,omitempty"`
+	Status ContainerRuntimeConfigStatus `json:"status,omitempty"`
+}
+
+// ContainerRuntimeConfigSpec defines the desired state of ContainerRuntimeConfig
+type ContainerRuntimeConfigSpec struct {
+	MachineConfigPoolSelector *metav1.LabelSelector          `json:"machineConfigPoolSelector,omitempty"`
+	ContainerRuntimeConfig    *ContainerRuntimeConfiguration `json:"containerRuntimeConfig,omitempty"`
+}
+
+// ContainerRuntimeConfiguration defines the tuneables of the container runtime
+type ContainerRuntimeConfiguration struct {
+	PidsLimit   int64  `json:"pidsLimit,omitempty"`
+	LogLevel    string `json:"logLevel,omitempty"`
+	LogSizeMax  int64  `json:"logSizeMax,omitempty"`
+	InfraImage  string `json:"infraImage,omitempty"`
+	OverlaySize string `json:"overlaySize,omitempty"`
```

Comment:
OverlaySize, LogSizeMax, and PidsLimit feel like they could be a resource.Quantity to allow users to nicely set the LogSize to say 1G. Thoughts?

Reply:
We can use it for `OverlaySize` and `LogSizeMax`. However, PidLimit is a number and best left as is.

Reply:
So use resource.Quantity and then convert them to strings? I don't see any validation happening in the Quantity to String function, so is changing the type really helpful in this case?

Reply:
Done, using resource.Quantity